oktapam/resource_database.go

```diff
@@ -0,0 +1,420 @@
+			Optional:    true,
+			Description: descriptions.DatabasePassword,
+			Sensitive:   true,
+		},
+		attributes.Secret: {
+			Type:        schema.TypeString,
+			Description: descriptions.DatabaseSecretID,
+			Computed:    true,
+		},
+	}}
+
+func resourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Description:   descriptions.ResourceDatabase,
+		CreateContext: resourceDatabaseCreate,
+		ReadContext:   resourceDatabaseRead,
+		UpdateContext: resourceDatabaseUpdate,
+		DeleteContext: resourceDatabaseDelete,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Computed: true,
+				// Description is autogenerated
+			},
+			attributes.ResourceGroup: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ResourceGroupID,
+			},
+			attributes.Project: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ProjectID,
+			},
+			attributes.CanonicalName: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.CanonicalName,
+			},
+			attributes.DatabaseType: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.DatabaseType,
+				ForceNew:    true,
+			},
+			attributes.RecipeBook: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.RecipeBookID,
+			},
+			attributes.ManagementConnectionDetails: {
+				Type:        schema.TypeList,
+				Required:    true,
+				MinItems:    1,
+				MaxItems:    1,
+				Description: descriptions.ManagementConnectionDetails,
+				Elem:        dbConnectionTypes,
+				ForceNew:    true,
+			},
+			attributes.ManagementGatewaySelector: {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: descriptions.ManagementGatewaySelector,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			attributes.ManagementGatewaySelectorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.ManagementGatewaySelectorID,
+			},
+			attributes.CreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.CreatedAt,
+			},
+			attributes.UpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.UpdatedAt,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDatabaseReadImport,
+		},
+	}
+}
+
+func resourceDatabaseCreate(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	c := getSDKClientFromMetadata(m)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+	canonicalName := d.Get(attributes.CanonicalName).(string)
+	dbType := d.Get(attributes.DatabaseType).(string)
+
+	var selectorLabels *map[string]string
+	if selector, ok := d.GetOk(attributes.ManagementGatewaySelector); ok {
```

Comment:
nit This logic we can pull out as utility method to convert tf map to go map[string]string.

Reply:
Moved this into a util func, `GetMapPtrFromResource`